### ISSUE TITLE
Better error message when System.NullReferenceException / IsNotMapped is thrown

### ISSUE THREAD

**Issue by techniq** [voronov-maxim/OdataToEntity#23]
Currently it is difficult to track down a `System.NullReferenceException` exception due to a`IsNotMapped` issue...
 
![image](https://user-images.githubusercontent.com/177476/42713264-d71d6636-86bc-11e8-8e1b-a1a1d80c836b.png)

Stacktrace:
```
   at OdataToEntity.EfCore.OeEfCoreEdmModelMetadataProvider.IsNotMapped(PropertyInfo propertyInfo) in /Users/techniq/Documents/Development/open-source/OdataToEntity/source/OdataToEntity.EfCore/OeEfCoreEdmModelMetadataProvider.cs:line 114
   at OdataToEntity.ModelBuilder.EntityTypeInfo.BuildProperties(Dictionary`2 entityTypes, Dictionary`2 enumTypes, Dictionary`2 complexTypes) in /Users/techniq/Documents/Development/open-source/OdataToEntity/source/OdataToEntity/ModelBuilder/EntityTypeInfo.cs:line 124
   at OdataToEntity.ModelBuilder.OeEdmModelBuilder.BuildEdmModel() in /Users/techniq/Documents/Development/open-source/OdataToEntity/source/OdataToEntity/ModelBuilder/OeEdmModelBuilder.cs:line 67
   at OdataToEntity.EfCore.OeEfCoreDataAdapterExtension.BuildEdmModelFromEfCoreModel(OeDataAdapter dataAdapter) in /Users/techniq/Documents/Development/open-source/OdataToEntity/source/OdataToEntity.EfCore/OeEfCoreDataAdapterExtension.cs:line 18
   at Finance.Web.Startup.ConfigureServices(IServiceCollection services) in /Users/techniq/Documents/Development/sbcs-chh/app-finance/Finance.Web/Startup.cs:line 81
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at Finance.Web.Program.Main(String[] args) in /Users/techniq/Documents/Development/sbcs-chh/app-finance/Finance.Web/Program.cs:line 10
```

I believe the issue is due to the model building being wrapped in a `try` and swallowing/not re-throwing the exception.
https://github.com/voronov-maxim/OdataToEntity/blob/06de2b2f639de46b869e9a0bbe334ee0a3f3ffd9/source/OdataToEntity.EfCore/OeEfCoreDataAdapterExtension.cs#L12-L23

It would be awesome if when a `IsNotMapped` is raised if more details were added, primarily what entity is not mapped (and maybe which entity is expecting it to be).  Thanks.

**Comment by voronov-maxim** [comment]
Could you please send your ef data context for reproduce this errror

**Comment by voronov-maxim** [comment]
Hi @techniq 
[1.6.0](https://www.nuget.org/packages/OdataToEntity/)

**Comment by techniq** [comment]
thanks @voronov-maxim, you're awesome!  I pulled it in and so far so good (I have a lot of controllers to port now, but my model builds now using a NuGet release 👍).

I wasn't able to create a simple model to reproduce the `IsNotMapped` issue like I did for #16 and #24.  I'm going to go ahead and close this issue for now unless I come across it again and can create a simple model.

It would be nice if we could enable verbose logging using [ILogger](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.1) at some point to help track down future model issues / get better insights into how the model is being built as right now I need to pull from source and setup project references instead of using the NuGet packages, and then sprinkle a good bit of `Debug.WriteLine` statements

Thanks again for the awesome work and quick turnaround on my issues.
